pkg/v22/resource/clusterconfigmap/resource.go

```diff
@@ -54,6 +58,9 @@ func New(config Config) (*StateGetter, error) {
 	if config.GetWorkerCountFunc == nil {
 		return nil, microerror.Maskf(invalidConfigError, "%T.GetWorkerCountFunc must not be empty", config)
 	}
+	if config.GetSubscriptionId == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.GetSubscriptionId must not be empty", config)
+	}
```

Comment:
We should have this error check but this means you need to add it to the AWS and KVM controllers as well.

Since its Azure specific the other controllers could return an empty string.